polaris/dataset/converters/_sdf.py

```diff
@@ -103,25 +103,31 @@ def _get_name(mol: dm.Mol):
         if self.mol_prop_as_cols:
             props &= ~Chem.PropertyPickleOptions.MolProps
         bytes_data = [mol.ToBinary(props) for mol in df[tmp_col]]
-
         df.drop(columns=[tmp_col], inplace=True)
 
-        # Create the zarr array
-        factory.zarr_root.array(self.mol_column, bytes_data, dtype=bytes, chunks=self.chunks)
+        if self.mol_column in factory.zarr_root and append:
```

Comment:
**Suggestion**: This could lead to confusing behavior when `append == True` but the column doesn't exist. In that case we deviate from what the user asked and do so silently. I think it would be better to just fail if that happens. 

```py
if append:  # Or: if axis == 1
    if self.mol_column not in factory.zarr_root:
        raise RuntimeError(...)
```

Reply:
If the column doesn't exist, it will create a new array. 
Here the `append` parameter is rather `allow_appending`. So the existing array is allowed to be appended when user set it to True. 
I can rename it to 'allow_appending' if that clarifies.